### PR TITLE
🏃 Remove capi-e2e tests

### PR DESCRIPTION
**Pull request by chuckha**
Signed-off-by: Chuck Ha <chuckh@vmware.com>

**What this PR does / why we need it**:
This PR removes the e2e tests. They are removed from test-infra and no longer used anywhere. The capd-e2es now replace the capi e2es entirely.

This PR also adds some make targets for common operations of using the capd-e2es. It provides 3 methods of running the tests

1) Generate manfiests, rebuild all provider images and run the tests
2) Rebuild the provider images and run the tests
3) Rebuild the docker provider and run the tests

/assign @wfernandes

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -105,12 +105,21 @@ test: ## Run tests
...
 
-.PHONY: test-e2e
-test-e2e: ## Run e2e tests
-	PULL_POLICY=IfNotPresent $(MAKE) docker-build
+.PHONY: full-capd-e2e
+full-capd-e2e: ## Rebuild all manifests and all provider images then run the capd-e2es
 	$(MAKE) generate-manifests
 	$(MAKE) release-manifests
-	cd ./test/e2e; MANAGER_IMAGE=$(CONTROLLER_IMG)-$(ARCH):$(TAG) go test -v -tags=e2e -timeout=1h . -args -ginkgo.v -ginkgo.trace
+	$(MAKE) all-images-capd-e2e
```
Comment (detiber) [inline Makefile]: Does `scripts/ci-capd-e2e.sh` need to be updated to run this?

Comment (chuckha) [reply Makefile]: no, unfortunately this part is a little confusing, but these are explicitly for dev-workflows. The CI script uses the lower-level methods (docker-build) with custom environment variables for CI

Comment (chuckha) [reply Makefile]: I do need to edit the docs though and the book to make the doc visible

File: Makefile [hunk 2]
```diff
@@ -105,12 +105,21 @@ test: ## Run tests
...
+
+.PHONY: all-images-capd-e2e
+all-images-capd-e2e: ## Rebuild all Cluster API provider images and run the capd-e2e tests
+	make docker-build REGISTRY=gcr.io/k8s-staging-cluster-api PULL_POLICY=IfNotPresent
+	$(MAKE) capd-e2e
+
+.PHONY: capd-e2e
+capd-e2e: ## Rebuild the docker provider and run the capd-e2e tests
```
Comment (wfernandes) [inline Makefile]: Just a suggestion but I think it would be more intuitive to do `make test-capd-e2e`

File: Makefile [hunk 3]
```diff
@@ -105,12 +105,21 @@ test: ## Run tests
 test-integration: ## Run integration tests
 	go test -v -tags=integration ./test/integration/...
 
-.PHONY: test-e2e
-test-e2e: ## Run e2e tests
-	PULL_POLICY=IfNotPresent $(MAKE) docker-build
+.PHONY: full-capd-e2e
```
Comment (wfernandes) [inline Makefile]: Just a suggestion but maybe it might be nice to prefix them with `test`.
E.g. `make test-capd-e2e-full`

File: Makefile [hunk 4]
```diff
@@ -105,12 +105,21 @@ test: ## Run tests
...
+full-capd-e2e: ## Rebuild all manifests and all provider images then run the capd-e2es
 	$(MAKE) generate-manifests
 	$(MAKE) release-manifests
-	cd ./test/e2e; MANAGER_IMAGE=$(CONTROLLER_IMG)-$(ARCH):$(TAG) go test -v -tags=e2e -timeout=1h . -args -ginkgo.v -ginkgo.trace
+	$(MAKE) all-images-capd-e2e
+
+.PHONY: all-images-capd-e2e
```
Comment (wfernandes) [inline Makefile]: `make test-capd-e2e-images`

**Comment by detiber** [comment]
lgtm, but will defer to @wfernandes for final lgtm

**Pull request merged**